### ISSUE TITLE
Drasil Research Group Meeting, Monday, Jan 22, 3:00 pm

### ISSUE THREAD

**Issue by smiths** [JacquesCarette/Drasil#3672]
Placeholder meeting agenda so that we don't forget to invite @cd155.

1. Introduction of @J1aM1ng
2. [Refined theories version of projectile](https://github.com/smiths/caseStudies/blob/master/CaseStudies/projectile/projectileSRS_RefinedTheories/Projectile_SRS.pdf)  (some information on our [Wiki](https://github.com/JacquesCarette/Drasil/wiki/Refined-Theories-version-of-Drasil)) (@smiths)

Regrets:

Virtual: @cd155, @samm82

**Comment by smiths** [comment]
Great summary @samm82!  Thank you for capturing the discussion.

We do need to think more about how we use rationale.  I agree that I was wrong to say it comes at the end.  Rationale for making decisions is important information throughout the process.  The challenge is that rationale is problem specific, but at the same time rationale can often be reused between different projects.  Some decisions can be justified when they are made, like the decision to use a Cartesian coordinate system, but other decisions are assumptions that restrict the applicability of the theory, but the rationale can be postponed.  For instance, the choice to have constant acceleration propagates through the theories.  The only time it really has to be justified is when someone chooses to use the theory to do something with it.  That means we can have generic theories that don't have a rationale yet, but do have a list of "proof obligations."  I don't really mean proof, just that the assumptions that need a rationale are identified.  I wonder if we can have assumptions/modelling decisions that may or may not have rationale attached to them.  The rationale can be postponed in some cases and provided in others.

I think our conversation showed that the noObstacles assumption can be justified in different ways.  Not having to worry about obstacles is there to simplify the problem.  It can be simpler in the sense that the physics for collisions doesn't have to be included, or in the sense that searching for collisions doesn't have to be done, or both.

The meeting is done, so I am going to close this issue.  If anyone wishes to add to the discussion though, please feel free to do so.  I'll use the discussion to add more content to the wiki entry related to refined theories, so that we can capture the information in a less transient location.

Yes, generating the relationship diagram would be great.  It is tedious to do by hand.  :smile:  I think real insight can be gained from pictures like this.  I didn't discuss it in the meeting today, but the entire branch based on TM:acceleration actually isn't necessary at all for the specific projectile problem we are solving.  We only need TM:velocity.  In hindsight this is obvious, since we are calculating using the position when the projectile is at ground level.  We don't need its velocity.  However, I didn't actually realize this until I made the picture.  :smile:  Of course, we don't have to actually draw the picture to do an analysis of the graph.  The final theories (Instance Models) are not actually connected to TM:acceleration through refinement.

The overall discussion reinforced for me that we really do want to be as general as possible.  We know we are allowing for the combination and transformation of theories, but we don't want to be too prescriptive on how people will use the theories.  We already saw many viewpoints on how to look at projectile motion, and if we talked to more people, we would see even more viewpoints.  All of that is for a relatively simple problem.  :smile:

We definitely want theories to be something that can be reused for different purposes in different contexts.

With respect to "Problem Specific Theories" versus "Projectile Specific Theories", I think we want the ability to use either approach.  A multiphysics example (maybe combining kinematics, kinetics and heat transfer) could have separate sections for the problem specific theories in each domain.  Having a label that reinforces the different topics could be helpful.  In other cases the author of the SRS may want to reinforce that they are following the template and have the more generic title "Problem Specific Theories."

**Comment by smiths** [comment]
I've thought further on rationales.  I think @balacij and @JacquesCarette are correct that rationales don't just come at the end.  Every scope decision, modelling decision, assumption and data constraint should get a rationale when created.  (For flexibility we can allow the rationales to be empty.  Maybe this would generate a warning when compiled?)   The rationale information can be overridden in favour of a new rationale.  For instance, the rationale for generic, non-problem specific, assumptions could have a generic rationale.  In the projectile example the first assumptions about constant acceleration could talk about the situations where constant acceleration arises, like for gravity and for a charged particle in an electric field. This generic rationale is overridden when a specific problem is chosen.

Our conversation also highlighted that we need a rationale for a sequence of theories.  The rationale isn't attached to a theory, but to a series of theories.  For example, the sequence from 3D accel, to 1D accel, to 2D accel in the projectile example is something that should be explained.  The explanation could go in the preamble to the section in the document, but we also should find a way to attach the rationale in the Drasil code as more than just text.

I've added these points to our next Drasil meeting so that we can discuss further.

**Comment by JacquesCarette** [comment]
Read all this - makes sense.

We should try hard to start implementing some of this. There are simple, infrastructural changes that are needed before we get into the heavy design of how all the above will work, and no reason to wait on starting those.

**Comment by smiths** [comment]
I'll add to the agenda for our next meeting to discuss the infrastructural changes, and who can be given ownership of those tasks.
